src/test/kotlin/no/nav/tiltakspenger/soknad/api/soknad/SøknadServiceTest.kt

```diff
@@ -0,0 +1,155 @@
+    private val mockPdfService = mockk<PdfService>().also { mock ->
+        coEvery { mock.lagPdf(any()) }
+    }
+    private val mockJoarkService = mockk<JoarkService>().also { mock ->
+        coEvery { mock.sendPdfTilJoark(any(), any(), any(), any()) }
+    }
+
+    private val gyldigSøknad = """
+        {
+          "tiltak": {
+            "aktivitetId": "123",
+            "søkerHeleTiltaksperioden": false,
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          },
+          "barnetillegg": {
+            "manueltRegistrerteBarnSøktBarnetilleggFor": [
+              {
+                "fornavn": "Test",
+                "etternavn": "Test",
+                "fødselsdato": "2025-01-01",
+                "bostedsland": "Test"
+              }
+            ],
+            "søkerOmBarnetillegg": true,
+            "registrerteBarnSøktBarnetilleggFor": [
+              {
+                "fornavn": "Test",
+                "fødselsdato": "2025-01-01",
+                "etternavn": "Testesen"
+              }
+            ],
+            "ønskerÅSøkeBarnetilleggForAndreBarn": true
+          },
+          "etterlønn": {
+            "mottarEllerSøktEtterlønn": true,
+            "utbetaler": "Test",
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          },
+          "institusjonsopphold": {
+            "borPåInstitusjon": true,
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          },
+          "introduksjonsprogram": {
+            "deltar": true,
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          },
+          "kvalifiseringsprogram": {
+            "deltar": true,
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          },
+          "pensjonsordning": {
+            "utbetaler": "Test",
+            "mottarEllerSøktPensjonsordning": true,
+            "periode": {
+              "fra": "2025-01-01",
+              "til": "2025-01-01"
+            }
+          }
+        }
+    """.trimMargin()
+
+    class MockMultiPartData(private val partDataList: MutableList<PartData>) : MultiPartData {
+        override suspend fun readPart(): PartData? {
+            return if (partDataList.isNotEmpty()) partDataList.removeLast() else null
+        }
+    }
+
+    private val søknadService = SøknadServiceImpl(
+        pdfService = mockPdfService,
+        joarkService = mockJoarkService,
+    )
+
+    @Test
+    fun `taInnSøknadSomMultipart leser inn MultiPartData med gyldig søknad to vedlegg`() {
+        val input: Input = mockk()
+        every { input.endOfInput } returns true
+        justRun { input.release() }
+        mockkStatic("no.nav.tiltakspenger.soknad.api.util.DetectKt")
```

Comment:
Hva skjer mellom linje 115-117 her? Spør ikke fordi jeg ikke liker koden, men fordi jeg ikke helt forstår den 😇

Reply:
Mocker to metoder fra input-mocken, siden vi uansett ikke skal ha noen innhold i vedleggs-filene i testen. Linje 117 trengs for å kunne mocke toplevel-metoden sjekkContentType() fra Detect.kt